source/Planet.h

```diff
@@ -37,6 +37,39 @@ class System;
 // are available, as well as attributes that determine what sort of missions
 // might choose it as a source or destination.
 class Planet {
+public:
+	class Port
+	{
+	public:
+		// The different properties that can be recharged by a port.
+		enum RechargeType
+		{
+			None = 0,
+			Shields = (1 << 0),
+			Hull = (1 << 1),
+			Energy = (1 << 2),
+			Fuel = (1 << 3),
+			All = Shields | Hull | Energy | Fuel,
```

Comment:
You're right now using RechargeType::All to determine whether a ship should be able to refill its crew if it's low. I'd prefer Crew its own recharge type.
```suggestion
			Crew = (1 << 4),
			All = Shields | Hull | Energy | Fuel | Crew,
```

Reply:
I was planning on making it depend on whether you can hire crew on the planet. I don't think it makes much sense to separate them.